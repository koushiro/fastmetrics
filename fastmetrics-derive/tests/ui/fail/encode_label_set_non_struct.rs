use fastmetrics_derive::EncodeLabelSet;

// This should fail because EncodeLabelSet can only be derived for structs
#[derive(EncodeLabelSet)]
enum NotAStruct {
    Variant1,
    Variant2,
}

fn main() {}
