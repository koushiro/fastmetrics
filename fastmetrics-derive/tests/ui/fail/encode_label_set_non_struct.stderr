error: #[derive(EncodeLabelSet)] can only be used for structs.
 --> tests/ui/fail/encode_label_set_non_struct.rs:5:6
  |
5 | enum NotAStruct {
  |      ^^^^^^^^^^
